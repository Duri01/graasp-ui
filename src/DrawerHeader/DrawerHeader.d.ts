import React from 'react';

interface DrawerHeaderProps {
  classes: {
    secondaryAction: string;
    drawerHeader: string;
  };
  theme: {
    direction?: string;
  };
  handleDrawerClose: () => {};
}

declare const DrawerHeader: React.FC<DrawerHeaderProps>;
