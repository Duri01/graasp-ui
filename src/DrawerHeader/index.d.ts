export { default } from './DrawerHeader';
export * from './DrawerHeader';
