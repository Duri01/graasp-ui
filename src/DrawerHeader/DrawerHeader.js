import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import IconButton from '@material-ui/core/IconButton';
import ListItem from '@material-ui/core/ListItem';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import ChevronRightIcon from '@material-ui/icons/ChevronRight';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import { DEFAULT_DIRECTION, DRAWER_HEADER_HEIGHT } from '../constants';

const styles = (theme) => ({
  drawerHeader: {
    height: DRAWER_HEADER_HEIGHT,
  },
  secondaryAction: {
    right: theme.spacing(1),
  },
});

export const DrawerHeader = ({
  classes,
  handleDrawerClose,
  children,
  theme: { direction = DEFAULT_DIRECTION },
}) => {
  return (
    <ListItem
      classes={{ root: classes.drawerHeader }}
      divider
      ContainerComponent='div'
    >
      {children}
      <ListItemSecondaryAction classes={{ root: classes.secondaryAction }}>
        <IconButton onClick={handleDrawerClose}>
          {direction === 'ltr' ? <ChevronLeftIcon /> : <ChevronRightIcon />}
        </IconButton>
      </ListItemSecondaryAction>
    </ListItem>
  );
};

const StyledComponent = withStyles(styles, { withTheme: true })(DrawerHeader);

export default StyledComponent;
