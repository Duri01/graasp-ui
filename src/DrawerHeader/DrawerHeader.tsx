import React, { FC } from 'react';
import { Theme, createStyles, makeStyles } from '@material-ui/core/styles';
import IconButton from '@material-ui/core/IconButton';
import ListItem from '@material-ui/core/ListItem';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import ChevronRightIcon from '@material-ui/icons/ChevronRight';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import { DEFAULT_DIRECTION, DRAWER_HEADER_HEIGHT } from '../constants';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    drawerHeader: {
      height: DRAWER_HEADER_HEIGHT,
    },
    secondaryAction: {
      right: theme.spacing(1),
    },
  }),
);

export interface DrawerHeaderProps {
  theme?: {
    direction?: string;
  };
  children?: JSX.Element;
  handleDrawerClose?: () => void;
}

export const DrawerHeader: FC<DrawerHeaderProps> = ({
  handleDrawerClose,
  children,
  theme,
}) => {
  const classes = useStyles();
  const direction = theme?.direction ?? DEFAULT_DIRECTION;
  return (
    <ListItem
      classes={{ root: classes.drawerHeader }}
      divider
      ContainerComponent='div'
    >
      {children}
      <ListItemSecondaryAction classes={{ root: classes.secondaryAction }}>
        <IconButton onClick={handleDrawerClose}>
          {direction === 'ltr' ? <ChevronLeftIcon /> : <ChevronRightIcon />}
        </IconButton>
      </ListItemSecondaryAction>
    </ListItem>
  );
};

export default DrawerHeader;
