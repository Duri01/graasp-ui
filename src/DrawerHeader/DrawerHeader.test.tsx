import React from 'react';
import IconButton from '@mui/material/IconButton';
import ListItem from '@mui/material/ListItem';
import ChevronLeftIcon from '@mui/icons-material/ChevronLeft';
import ChevronRightIcon from '@mui/icons-material/ChevronRight';
import { ShallowWrapper, shallow } from 'enzyme';
import { DrawerHeader, DrawerHeaderProps } from './DrawerHeader';

const createDrawerHeaderProps = (opts?: {
  direction: string;
}): DrawerHeaderProps => {
  return {
    theme: {
      direction: opts?.direction,
    },
    handleDrawerClose: jest.fn(),
  };
};

describe('DrawerHeader', () => {
  let wrapper: ShallowWrapper;

  describe('default values', () => {
    beforeAll(() => {
      const props = createDrawerHeaderProps();
      // eslint-disable-next-line react/jsx-props-no-spreading
      wrapper = shallow(<DrawerHeader {...props} />);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('renders one <ListItem /> component with correct chevron icon', () => {
      const listItem = wrapper.find(ListItem);
      expect(listItem).toHaveLength(1);
      const icon = listItem.find(IconButton);
      expect(icon).toHaveLength(1);
      const menuIcon = icon.find(ChevronLeftIcon);
      expect(menuIcon).toHaveLength(1);
    });
  });

  describe('prop values and children', () => {
    const children = <div className='test-child'>some child</div>;

    beforeAll(() => {
      const props = createDrawerHeaderProps({ direction: 'rtl' });
      // eslint-disable-next-line react/jsx-props-no-spreading
      wrapper = shallow(<DrawerHeader {...props}>{children}</DrawerHeader>);
    });

    it('renders one <ListItem /> component with correct chevron icon', () => {
      const listItem = wrapper.find(ListItem);
      expect(listItem).toHaveLength(1);
      const childrenComponent = wrapper.find('.test-child');
      expect(childrenComponent).toHaveLength(1);
      expect(childrenComponent.text()).toEqual('some child');
      const icon = listItem.find(IconButton);
      expect(icon).toHaveLength(1);
      const menuIcon = icon.find(ChevronRightIcon);
      expect(menuIcon).toHaveLength(1);
    });
  });
});
