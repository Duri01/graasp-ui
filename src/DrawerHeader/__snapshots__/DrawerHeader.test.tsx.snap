// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerHeader default values renders correctly 1`] = `
<WithStyles(ForwardRef(ListItem))
  ContainerComponent="div"
  classes={
    Object {
      "root": "makeStyles-drawerHeader-1",
    }
  }
  divider={true}
>
  <WithStyles(ForwardRef(ListItemSecondaryAction))
    classes={
      Object {
        "root": "makeStyles-secondaryAction-2",
      }
    }
  >
    <WithStyles(ForwardRef(IconButton))
      onClick={[MockFunction]}
    >
      <ChevronLeftIcon />
    </WithStyles(ForwardRef(IconButton))>
  </WithStyles(ForwardRef(ListItemSecondaryAction))>
</WithStyles(ForwardRef(ListItem))>
`;
